### PR TITLE
Added support for full text search

### LINKED ISSUE #5755: Expose minimal full text search

### PR DESCRIPTION
**Pull request by papafe**
This closes #5755

Depends on updating core to the latest release (https://github.com/realm/realm-js/pull/5796). The update has already been merged in this branch.

## ☑️ ToDos
* [x] 📝 Changelog entry
* [x] 🚦 Tests

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,7 +7,11 @@
...
 * Enable multiple processes to operate on an encrypted Realm simultaneously. ([realm/realm-core#1845](https://github.com/realm/realm-core/issues/1845))
 * Added `Realm.setLogger`, that allows to setup a single static logger for the duration of the app lifetime. Differently from the now deprecated sync logger (that was setup with `Sync.setLogger`), this new one will emit messages coming also from the local database, and not only from sync. It is also possible to change the log level during the whole duration of the app lifetime with `Realm.setLogLevel`. ([#2546](https://github.com/realm/realm-js/issues/2546))
+* Added support for a sync configuration option to provide an `SSLConfiguration` with a custom function for validating the server's SSL certificate. [#5485](https://github.com/realm/realm-js/issues/5485)
+* Extended `PropertySchema.indexed` with the `fulltext` option, that allows to create an index for full-text search queries.  [#5755](https://github.com/realm/realm-js/issues/5755)
```
Comment (kneth) [inline CHANGELOG.md:11]:
```suggestion
* Extended `PropertySchema.indexed` with the `fulltext` option, that allows to create an index for full-text search queries.  ([#5755](https://github.com/realm/realm-js/issues/5755))
```

**Review by kneth — APPROVED** [review]
Please create an issue to add support for the new index type in the Babel plugin.

**Review by kraenhansen — APPROVED** [review]

File: packages/realm/src/schema/to-binding.ts [hunk 2]
```diff
@@ -99,10 +99,16 @@ export function toBindingPropertySchema(name: string, schema: CanonicalPropertyS
   const result: BindingProperty = {
     name,
     type: toBindingPropertyType(schema),
-    isIndexed: schema.indexed,
     objectType: schema.objectType && schema.objectType in TYPE_MAPPINGS ? undefined : schema.objectType,
     linkOriginPropertyName: schema.property,
   };
+
+  if (schema.indexed === "fulltext") {
```
Comment (takameyer) [inline packages/realm/src/schema/to-binding.ts:106]: Is `fulltext` a single word?  I would expect this to be either `full-text` or `fullText`.

File: packages/realm/src/schema/types.ts [hunk 3]
```diff
@@ -84,14 +84,21 @@ export type CanonicalRealmSchema = CanonicalObjectSchema[];
  */
 export type CanonicalObjectSchemaProperty = CanonicalPropertySchema;
 
+/**
+ * The type of index on a property:
+ * - 'true' enables a regular index
+ * - `fulltext` enables a full-text search index and can only be applied to string properties.
```
Comment (elle-j) [inline packages/realm/src/schema/types.ts:90]:
```suggestion
 * - `true` enables a regular index.
 * - `"fulltext"` enables a full-text search index and can only be applied to string properties.
```

File: packages/realm/src/schema/validate.ts [hunk 4]
```diff
@@ -164,7 +164,10 @@ export function validatePropertySchema(
       assert.string(property, `'${propertyName}.property' on '${objectName}'`);
     }
     if (indexed !== undefined) {
-      assert.boolean(indexed, `'${propertyName}.indexed' on '${objectName}'`);
+      assert(
+        typeof indexed === "boolean" || indexed === "fulltext",
+        `"indexed" needs to be either a boolean or "fulltext"`,
```
Comment (elle-j) [inline packages/realm/src/schema/validate.ts:169]: The change of error message should have caused [this validation test](https://github.com/realm/realm-js/blob/d9df13ff87a4c26e8e2eac188927ac4873ebc37f/packages/realm/src/tests/schema-validation.test.ts#L233C5-L240) to fail. You can run from the root `npm test --workspace realm`. Even with this change, the test should fail but the test need to be updated by changing the last part of the expected error message. (Can make a small PR against this branch to add minor things.)
```suggestion
        `Expected '${propertyName}.indexed' on '${objectName}' to be a boolean or 'fulltext'.`,
```

File: integration-tests/tests/src/tests/queries.ts [hunk 5]
```diff
@@ -131,6 +150,65 @@ const expectQueryResultValues = (
 };
 
 describe("Queries", () => {
+  describe("Full text search", () => {
```
[the remark this answers is not in the document]
Comment (papafe) [reply integration-tests/tests/src/tests/queries.ts:153]: Added

File: packages/realm/src/schema/to-binding.ts (hunk 2 above)
Comment (papafe) [reply packages/realm/src/schema/to-binding.ts:106]: Usually it's written as "full-text". I didn't think too much about it, but I think it makes sense to write it like that

**Review by elle-j — COMMENTED** [review]
Great new feature support 🙂 Left some comments. There are minor updates to things not included in this PR that I can add in another PR that targets this branch.

**Review by elle-j — APPROVED** [review]
Looks awesome! The only thing you may want to wait with is the spec changes. You could coordinate with Mathias on that 👍

File: packages/realm/src/schema/validate.ts [hunk 6]
```diff
@@ -149,6 +148,7 @@ export function validatePropertySchema(
   objectName: string,
   propertyName: string,
   propertySchema: unknown,
+  primaryKey: string | undefined,
```
Comment (papafe) [inline packages/realm/src/schema/validate.ts:151]: Does this make sense? @kraenhansen @elle-j @takameyer

Comment (elle-j) [reply packages/realm/src/schema/validate.ts:151]: Currently we have the validation part (i.e. `validate.ts`) be responsible for checking that the JS data types are correct. E.g. checking that `objectType` is a string, but not checking that `objectType` only is set when `type` is e.g. `"list"` (that's done in `normalize.ts`). So the validation is just for making sure that other functions can rely on the types being what TS says it is.

So maybe we can move the logic of checking that the combination of `"full-text"` and other props are allowed, to `normalize.ts`?

Comment (papafe) [reply packages/realm/src/schema/validate.ts:151]: Makes total sense!

**Pull request merged**